src/App.jsx

```diff
@@ -3,10 +3,12 @@ import './App.css'
 import Layout from './views/Layout/Layout'
 import Home from './views/Home/Home'
 import { fetchUser } from './services/user'
+import { useUsers } from './context/UserContext'
 
 function App() {
   // inital value should match the data type of end value
-  const [user, setUser] = useState({})
+  // const [user, setUser] = useState({})
+  const {user, setUser} = useUsers()
 
   useEffect(() => {
```

Comment:
If you move your useEffect inside of your UserProvider inside the UserContext file, you can control the state from that single place without worrying about passing that piece of state around